app/src/main/java/org/oppia/app/parser/StringToRatioParser.kt

```diff
@@ -14,7 +15,7 @@ class StringToRatioParser {
   private val invalidCharsRegex =
     """^[\d\s/:]+$""".toRegex()
   private val invalidRatioRegex =
-    """^(\s)*(\d+((\s)*:(\s)*\d+)+)(\s)*$""".toRegex()
+    """^(\s)?\d+(\s)?(:(\s)?\d+(\s)?)+(\s)?$""".toRegex()
```

Comment:
```suggestion
    """^\d+\s?(:\s?\d+\s?)+$""".toRegex()
```

Normalize also trims whitespace from the start & end.

Reply:
Doesn't it replaces spaces with greater than length 2 by one space ?

Reply:
done